src/main/java/com/sda/microblogging/service/UserService.java

```diff
@@ -46,6 +49,20 @@ public User updateUserPassword(@NotBlank Integer userId, @NotBlank @Size(min = 1
         }
     }
 
+    public User updateUserPrivacy(@NotBlank String userEmail, boolean isPrivate) {
```

Comment:
I think this whole thing can be converted into these line. I think this is how Ahmad said to handle the optional. He did not recommend to use ```user.get()``` to access an optional user. 
And I don't know if we need separate exception for UserAlreadyHasRequestedPrivacyException. I think it should be a toggle instead? 

Because if the user is private and we give it again a ``` true ``` boolean value, it will store the same value, but may be hibernate will handle that since nothing change??? I don't now.

```
User user = userRepository.findByEmail(userEmail)
                           .orElseThrow(UserNotFound::new);

user.setPrivate(isPrivate);
return userRepository.save(user);
```

Reply:
I agree, UserAlreadyHasRequestedPrivacyException can be simplified, no really need in it (but only not toggle for privacy change: will set it, more secure).